### DISCUSSION TITLE
How to disable check for ruff in the virtual envs site-packages?

### DISCUSSION THREAD

**Discussion by jiaulislam** [astral-sh/ruff#5509] [Q&A] [answered]
I don't know why but ruff is checking my virtuals envs packages also for linting or errors which is for my case not wanted or I don't know if anybody does. Is there any way to make it stop ?

> Ruff Config in `pyproject.toml` file:
```
[tool.ruff]
select = [
    "E",  # pycodestyle errors
    "W",  # pycodestyle warnings
    "F",  # pyflakes
    "I",  # isort
    "C",  # flake8-comprehensions
    "B",  # flake8-bugbear
]
ignore = [
    "E501",  # line too long, handled by black
    "B008",  # do not perform function calls in argument defaults
    "C901",  # too complex
]
exclude = [
    ".bzr",
    ".direnv",
    ".eggs",
    ".git",
    ".git-rewrite",
    ".hg",
    ".mypy_cache",
    ".nox",
    ".pants.d",
    ".pytype",
    ".ruff_cache",
    ".svn",
    ".tox",
    ".venv",
    "__pypackages__",
    "_build",
    "buck-out",
    "build",
    "dist",
    "node_modules",
    "venv",
    "./epay/tests"
]

[tool.isort]
known-third-party = ["fastapi", "python-dateutil", "python-dotenv", "oracledb", "num2words", "httpx", "pytz", "bcrypt"]
```

TIA

![image](https://github.com/astral-sh/ruff/assets/54698236/7bb6502e-a5c0-4d7a-badf-8fbdde3e3767)

**Comment by charliermarsh** [comment, marked as answer]
Can you add `.pyenv` as an exclusion?

```toml
exclude = [
    ".bzr",
    ".direnv",
    ".eggs",
    ".git",
    ".git-rewrite",
    ".hg",
    ".mypy_cache",
    ".nox",
    ".pants.d",
    ".pytype",
    ".ruff_cache",
    ".svn",
    ".tox",
    ".venv",
    "__pypackages__",
    "_build",
    "buck-out",
    "build",
    "dist",
    "node_modules",
    "venv",
    "./epay/tests",
    ".pyenv"
]
```

We can also add it to our defaults.

**Reply by jiaulislam** [reply]
Thanks it solved the issue.

**Reply by charliermarsh** [reply]
If you want, you can also put your custom stuff in `extend-exclude` so that you don't have to repeat the defaults, e.g., replace `exclude = [...]` with:

```toml
extend-exclude = [
    "./epay/tests",
    ".pyenv",
]
```

**Reply by jiaulislam** [reply]
that would be wise and neat'n clean i guess. thanks @charliermarsh

**Reply by jiaulislam** [reply]
@charliermarsh hi again, I don't know whats happening but It seems that ruff is checking the site-packages directory even after providing ".pyenv" in exclude or extend-exclude block. Sometimes it works(if I restart my vscode it works !) but sometime its not.

![image](https://github.com/astral-sh/ruff/assets/54698236/785b4238-6a8c-4141-b343-593adefc0c44)

**Comment by jiaulislam** [comment]
For those who are using ruff plugin for vscode and trying to solve this problem. Try to set the pyproject file for the plugin as below snapshot.

![image](https://github.com/astral-sh/ruff/assets/54698236/7f27c33a-70d0-47a1-997d-77a646122d15)

**Comment by 573dave** [comment]
I use binhex-code-server on unraid and adding this arg in the extension settings fixed the ruff "Problems" for me:
```
--exclude=.venv/*
```
